alembic/versions/7d2b9fb924fa_add_sync_release_pull_request_table.py

```diff
@@ -0,0 +1,194 @@
+    )
+
+    @classmethod
+    def get_or_create(
+        cls,
+        pr_id: int,
+        namespace: str,
+        repo_name: str,
+        project_url: str,
+        session: sa.orm.Session,
+    ) -> "SyncReleasePullRequestModel":
+        project = GitProjectModel.get_or_create(
+            namespace=namespace,
+            repo_name=repo_name,
+            project_url=project_url,
+            session=session,
+        )
+        pr = (
+            session.query(SyncReleasePullRequestModel)
+            .filter_by(pr_id=pr_id, project_id=project.id)
+            .first()
+        )
+        if not pr:
+            pr = SyncReleasePullRequestModel()
+            pr.pr_id = pr_id
+            pr.project_id = project.id
+            session.add(pr)
+        return pr
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "sync_release_pull_request",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("pr_id", sa.Integer(), nullable=True),
+        sa.Column("project_id", sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["project_id"],
+            ["git_projects.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(
+        op.f("ix_sync_release_pull_request_pr_id"),
+        "sync_release_pull_request",
+        ["pr_id"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_sync_release_pull_request_project_id"),
+        "sync_release_pull_request",
+        ["project_id"],
+        unique=False,
+    )
+    op.add_column(
+        "sync_release_run_targets",
+        sa.Column("downstream_pr_id", sa.Integer(), nullable=True),
+    )
+    op.create_foreign_key(
+        None,
+        "sync_release_run_targets",
+        "sync_release_pull_request",
+        ["downstream_pr_id"],
+        ["id"],
+    )
+
+    # ### end Alembic commands ###
+    bind = op.get_bind()
+    session = sa.orm.Session(bind=bind)
+
+    # Create Packit downstream pull request models
+    # and missing git projects.
+
+    # Split the groups back, this may not fully produce the same thing.
+    for sync_release in session.query(SyncReleaseTargetModel):
+        if sync_release.downstream_pr_url and not sync_release.pull_request:
+            url = sync_release.downstream_pr_url
+            # noqa[203]: prettier like it this way
+            project_url = url[0 : (url.rfind("/pull-request/"))]  # noqa[203]
+            pr_id = int(url[(url.rfind("/pull-request/") + 14) :])  # noqa[203]
+            namespace = "rpms"
+            repo = url[
+                (url.rfind("rpms/") + 5) : url.rfind("/pull-request")  # noqa[203]
+            ]
+            pull_request = SyncReleasePullRequestModel.get_or_create(
+                pr_id, namespace, repo, project_url, session
+            )
+            sync_release.pull_request = pull_request
+            session.add(pull_request)
```

Comment:
nice that we will have this for the past data! 

I would maybe add some error handling in case some of the URLs would be malformed, but if you have tested this with prod dump and it works, then no need

Reply:
Yes, I have tested it with the today db and it seems to work. But still I can handle errors 👍